modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -547,6 +590,11 @@ RelDataType syncAttributes(
             return null;
         }
 
+        if ((SqlTypeUtil.isIntType(type1) && SqlTypeUtil.isCharacter(type2))
```

Comment:
This is not consistent with the following coercion rules used for columns (see `AbstractTypeCoercion::commonTypeForBinaryComparison`):

> // 1 > '1' will be coerced to 1 > 1.
    if (SqlTypeUtil.isAtomic(type1) && SqlTypeUtil.isCharacter(type2)) 
       return type1;

>   if (SqlTypeUtil.isAtomic(type2) && SqlTypeUtil.isCharacter(type1)) 
       return type2

I think this if-clause is unnecessary.